composable_kernel/include/tensor_operation/reduction_functions_blockwise.hpp

```diff
@@ -23,247 +23,161 @@
  * SOFTWARE.
  *
  *******************************************************************************/
-#ifndef CK_REDUCTION_FUNCTIONS_BLOCKWISE_HPP
-#define CK_REDUCTION_FUNCTIONS_BLOCKWISE_HPP
+#ifndef CK_REDUCTION_FUNCTIONS_PARTITIONED_BLOCKWISE_HPP
+#define CK_REDUCTION_FUNCTIONS_PARTITIONED_BLOCKWISE_HPP
 
 #include "data_type.hpp"
 
 #include "reduction_common.hpp"
 #include "reduction_operator.hpp"
-#include "reduction_functions_binop.hpp"
+#include "reduction_functions_accumulate.hpp"
 
 namespace ck {
 
-template <typename buffer2dDescType,
-          bool blockIsOneRow,
+template <typename buffer1dDescType,
```

Comment:
typename BufferDesc_M or maybe BufferDesc_K?

Reply:
Logically, this actually include both M and K,  but could be M_K or K_M.   It will be updated later

Reply:
In next PR, Please use a fixed LDS layout [M, K] or [K, M], so you don't need to support both logic.

Reply:
Ok.    Can I use a lamba to generate a layout which is either [M,K] or [K,M], so that both logic can be supported ?  The if constexpr is used inside the lamba to make choice

Reply:
Please don't... Just use [M, K], I think it's better for performance. We can talk about it if you are not sure

Reply:
Yes, the buffer descriptor for LDS should always  use `[M, K]`,  but the thread cluster lay out could be `[M, K]` or `[K, M]` depending on the vector dim

Reply:
Let's talk.